src/bp/core/services/dialog/flow/service.ts

```diff
@@ -66,20 +120,31 @@ export class FlowService {
   }
 
   private _listenForCacheInvalidation() {
-    this.cache.events.on('invalidation', (key: string) => {
-      const matches = key.match(/\/bots\/([A-Z0-9-_]+)\/flows\//i)
-      if (matches && matches.length >= 1) {
+    this.cache.events.on('invalidation', async (key: string) => {
+      if (this._flowCache.empty()) {
+        return
+      }
```

Comment:
This will invalidate the cache when any file is invalidated, which makes the cache pretty much useless

Reply:
My bad, I should rename the function to isEmpty since it returns whether the flowCache internal map size is of 0.